[CMake] FIX Debug build + more cleaning

guparan opened the pull request:
3 changes here:
- ~~FIX Debug build --> force all Zlib links to use the Release version~~
- CLEAN SOFA_HAVE_SOFA.YYY macro for consistency --> use SOFA_HAVE_SOFA_YYY everywhere
- FIX Boost::program_option copy --> `sofa_install_libraries` was missing

**IMPORTANT:** Windows users must update the `lib` directory in SOFA sources with https://www.sofa-framework.org/download/WinDepPack/latest/

Fixes #1851

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

guparan committed:
[CMake] Force all Zlib links to use Release version of the lib

guparan committed:
[CMake] FIX Qt libs copy in Debug with windeployqt

guparan committed:
[CMake] FIX: use XXX_HAVE_SOFA_YYY instead of XXX_HAVE_SOFA.YYY

Also fix missing Boost::program_option copy on Windows

jnbrunet commented:
Shouldn't be `SOFA_HAS_XXXX` ?

https://www.thesaurus.com/e/grammar/has-vs-have/

guparan commented:
Historical reasons ... you know the drill :disappointed_relieved: 
Consistency is more important than grammatical accuracy :+1:

fredroy commented:
seems that CMake uses HAVE_*
(see https://cmake.org/cmake/help/latest/module/CheckSymbolExists.html or https://cmake.org/cmake/help/latest/module/CheckTypeSize.html ) 🧐

guparan commented:
Back to WIP ... my fix was not a fix :+1: :+1: :+1:

guparan committed:
[CMake] Remove calls to sofa_set_targets_release_only

Windows users: update the WinDepPack to fix the Debug build.
https://www.sofa-framework.org/download/WinDepPack/latest/

guparan commented:
Ok it's fixed now. Windows users must update the `lib` directory in SOFA sources with https://www.sofa-framework.org/download/WinDepPack/latest/ :+1:

fredroy committed:
Merge branch 'master' into fix_debug_build

fredroy commented on applications/plugins/CImgPlugin/CMakeLists.txt:
```diff
@@ -88,6 +84,8 @@ sofa_create_package_with_targets(
     RELOCATABLE "plugins"
     )
 
+debug_print_target_properties(${PROJECT_NAME})
```
Intentional ?

guparan replied:
Nope :laughing:

guparan committed:
[CImgPlugin] Remove debug log in CMakeLists
